### PR TITLE
[deep link] [ios] Add a dropdown to choose ios target (default is `runner`)

### PR DESCRIPTION
**Pull request by hangyujin**
image

issue: https://github.com/flutter/devtools/issues/7799
## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [ ] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or there is a reason for not adding tests.

![build.yaml badge]

If you need help, consider asking for help on [Discord].

[Contributor Guide]: https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[build.yaml badge]: https://github.com/flutter/devtools/actions/workflows/build.yaml/badge.svg

### CODE REVIEW

File: packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart [hunk 1]
```diff
@@ -45,13 +45,19 @@ class _DeepLinkListViewState extends State<DeepLinkListView>
...
+      controller.selectedAndroidVariantIndex.value = _getDefaultVariantIndex(
         controller.selectedProject.value!.androidVariants,
+        defaultVariant: 'release',
       );
       if (FeatureFlags.deepLinkIosCheck) {
         controller.selectedIosConfigurationIndex.value =
-            _getReleaseVariantIndex(
+            _getDefaultVariantIndex(
           controller.selectedProject.value!.iosBuildOptions.configurations,
+          defaultVariant: 'release',
```
Comment (chunhtai) [inline packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart:56]: defaultConfiguration

File: packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart [hunk 2]
```diff
@@ -73,9 +79,12 @@ class _DeepLinkListViewState extends State<DeepLinkListView>
     );
   }
 
-  int _getReleaseVariantIndex(List<String> variants) {
+  int _getDefaultVariantIndex(
```
Comment (chunhtai) [inline packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart:82]: this probably shouldn't be called variant anymore

Comment (chunhtai) [reply packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart:82]: same for other places in the code that are shared between ios and android.

File: packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart [hunk 3]
```diff
@@ -307,24 +326,21 @@ class _DeepLinkListViewTopPanel extends StatelessWidget {
...
   const _VariantDropdown({
-    required this.os,
-    required this.controller,
+    required this.valuenotifier,
+    required this.variants,
+    required this.title,
   });
-  final PlatformOS os;
-  final DeepLinksController controller;
+  final ValueNotifier<int> valuenotifier;
```
Comment (kenzieschmoll) [inline packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart:333]: nit: rename to notifier

File: packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart (hunk 2 above)
Comment (hangyujin) [reply packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart]: renamed to configuration

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**